code/controllers/subsystem/eigenstate.dm

```diff
@@ -0,0 +1,109 @@
+///Subsystem used to teleport people to a linked web of itterative entries. If one entry is deleted, the 2 around it will forge a link instead.
+SUBSYSTEM_DEF(eigenstates)
+	name = "Eigenstates"
+	flags = SS_NO_INIT | SS_NO_FIRE
+	///The list of objects that something is linked to indexed by UID
+	var/list/eigen_targets = list()
+	///UID to object reference
+	var/list/eigen_id = list()
+	///Unique id counter
+	var/id_counter = 1
+	///Limit the number of sparks created when teleporting multiple atoms to 1
+	var/spark_time = 0
+
+///Creates a new link of targets unique to their own id
+/datum/controller/subsystem/eigenstates/proc/create_new_link(targets)
+	if(length(targets) <= 1)
+		return FALSE
+	for(var/atom/target as anything in targets) //Clear out any connected
+		var/already_linked = eigen_id[target]
+		if(!already_linked)
+			continue
+		if(length(eigen_targets[already_linked]) > 1) //Eigenstates are notorious for having cliques!
+			target.visible_message("[target] fizzes, it's already linked to something else!")
+			targets -= target
+			continue
+		target.visible_message("[target] fizzes, collapsing it's unique wavefunction into the others!") //If we're in a eigenlink all on our own and are open to new friends
+		remove_eigen_entry(target) //clearup for new stuff
+	//Do we still have targets?
+	if(!length(targets))
+		return FALSE
+	var/atom/visible_atom = targets[1] //The object that'll handle the messages
+	if(length(targets) == 1)
+		visible_atom.visible_message("[targets[1]] fizzes, there's nothing it can link to!")
+		return FALSE
+
+	eigen_targets["[id_counter]"] = list() //Add to the master list
+	for(var/atom/target as anything in targets)
+		eigen_targets["[id_counter]"] += target
+		eigen_id[target] = "[id_counter]"
+		RegisterSignal(target, COMSIG_CLOSET_INSERT, .proc/use_eigenlinked_atom)
+		RegisterSignal(target, COMSIG_PARENT_QDELETING, .proc/remove_eigen_entry)
+		RegisterSignal(target, COMSIG_ATOM_TOOL_ACT(TOOL_WELDER), .proc/tool_interact)
+		var/obj/item = target
+		if(item)
+			item.color = "#9999FF" //Tint the locker slightly.
+			item.alpha = 200
+			do_sparks(3, FALSE, item)
+
+	visible_atom.visible_message("The items' eigenstates spilt and merge, linking each of them together.")
```

Comment:
This is a weird visual message in that it's not something you'd be able to see normally (I don't know what an eigenstate is, but I'm assuming it's more a state of something rather than something physical). Can you replace this with something that gives off the same idea (that something of interest has happened), but more in line with something you could actually see?

Reply:
I went through the comments and tried to make it more in line with what you're asking